File: src/main/java/burrows/apps/example/gif/presentation/di/module/GlideModule.java
package burrows.apps.example.gif.presentation.di.module;

import android.content.Context;
import burrows.apps.example.gif.data.rest.repository.ImageRepository;
import burrows.apps.example.gif.presentation.di.scope.PerActivity;
import dagger.Module;
import dagger.Provides;

/**
 * @author <a href="mailto:jaredsburrows@gmail.com">Jared Burrows</a>
 */
@Module
public class GlideModule {
  @Provides @PerActivity protected ImageRepository provideImageDownloader(Context context) {
    return new ImageRepository(context);
  }
}

File: src/main/kotlin/burrows/apps/example/gif/presentation/di/module/GlideModule.kt
package burrows.apps.example.gif.presentation.di.module

import android.content.Context
import burrows.apps.example.gif.data.rest.repository.ImageApiRepository
import burrows.apps.example.gif.presentation.di.scope.PerActivity
import dagger.Module
import dagger.Provides

/**
 * @author [Jared Burrows](mailto:jaredsburrows@gmail.com)
 */
@Module
open class GlideModule {
  @Provides @PerActivity open fun providesImageDownloader(context: Context): ImageApiRepository {
    return ImageApiRepository(context)
  }
}
